src/Telegram/Types/Chat/ChatMemberResolver.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace SergiX44\Nutgram\Telegram\Types\Chat;
+
+use Attribute;
+use Exception;
+use SergiX44\Hydrator\Annotation\ConcreteResolver;
+
+#[Attribute(Attribute::TARGET_CLASS)]
+class ChatMemberResolver extends ConcreteResolver
+{
+    public function getConcreteClass(array $data): string
+    {
+        return match ($data['status']) {
+            'creator' => ChatMemberOwner::class,
+            'administrator' => ChatMemberAdministrator::class,
+            'member' => ChatMemberMember::class,
+            'restricted' => ChatMemberRestricted::class,
+            'left' => ChatMemberLeft::class,
+            'kicked' => ChatMemberBanned::class,
+            default => throw new Exception('Unknown ChatMember status: '.$data['status']),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Maybe an illegal argument exception is more appropriate to the context